pkg/daemon/daemon.go

```diff
@@ -554,6 +613,245 @@ func (dn *Daemon) detectEarlySSHAccessesFromBoot() error {
 	return nil
 }
 
+// RunHypershift is the entry point for the simplified Hypershift mode daemon
+func (dn *Daemon) RunHypershift(stopCh <-chan struct{}, exitCh <-chan error) error {
+	glog.Info("Starting MachineConfigDaemon - Hypershift")
+
+	signaled := make(chan struct{})
+	dn.InstallSignalHandler(signaled)
+
+	defer utilruntime.HandleCrash()
+	defer dn.queue.ShutDown()
+
+	go wait.Until(dn.worker, time.Second, stopCh)
+
+	for {
+		select {
+		case <-stopCh:
+			return nil
+		case <-signaled:
+			return nil
+		case err := <-exitCh:
+			// This channel gets errors from auxiliary goroutines like loginmonitor and kubehealth
+			// TODO we really shouldn't have any for hypershift
+			glog.Warningf("Got an error from auxiliary tools: %v", err)
+		}
+	}
+}
+
+func (dn *Daemon) syncNodeHypershift(key string) error {
+	// First, get the current and desired configurations for the node
+	// current configuration will be read from on-disk state, either
+	//   a) /etc/mcd-currentconfig.json, written by a previous hypershift-mode MCD
+	//   b) /etc/mcs-machine-config-content.json, written by MCS when the node is provisioned,
+	//      if no MCD has operated on this node
+	// desired configuration will be read directly off a ConfigMap in our namespace, specified by
+	// dn.configMap. This currently has a "config" key (full ignition served json) and a "hash"
+	// key, which is the TargetVersionConfigHash for Hypershift nodepools
+
+	// This isn't strictly necessary but we should only react to our own node changes, like normal MCD
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	if name != dn.name {
+		return nil
+	}
+
+	mcsServedConfigPath := "/etc/mcs-machine-config-content.json"
```

Comment:
Do you expect these to change, if not, const

Reply:
Ah I think this is already a const somewhere. Will modify